### PR TITLE
Adding first game

### PR DESCRIPTION
**Pull request by chompasina**
Pahlka's Friday Spike for 11/3: We added a new game constructor which will keep track of our character and check for collision detection on our portal. Inside of the Game constructor we're now cycling our character to the other side of the screen. We also added a checkPortal function (WIP) that will eventually check different game states.

### CODE REVIEW

File: lib/game.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+require ('./character')
+
+function Game(character){
+  this.character = character;
+}
+
+Game.prototype.checkEdge = function(){
+  if(this.character.x > 600){
```
Comment (deborahleehamel) [inline lib/game.js:8]: Cool to see how the character can travel off the canvas on the right side and also re-enter from the left side as a user might expect.

File: lib/index.js [hunk 2]
```diff
@@ -11,12 +12,23 @@ function drawCanvas(){
...
   context.fillRect(character.x, character.y, character.width, character.height);
 }
 
+function drawPortal(){
+  context.fillStyle = "papayawhip";
+  context.fillRect(500, 0, 100, 100);
+}
+
+function checkPortal(){
+  //conditional for game state: if game.state===2, make new game-- render new canvas,
```
Comment (deborahleehamel) [inline lib/index.js:28]: We can give each new game an indexed number to match the state that gets called when our character enters/collides with each portal. We can add as many new portals/games as Pahlka would like to imagine/create.

File: lib/index.js [hunk 3]
```diff
@@ -11,12 +12,23 @@ function drawCanvas(){
...
 
 let character = new Character();
+let game = new Game(character);
 
 function drawSprite(){
   context.fillStyle = "green";
   context.fillRect(character.x, character.y, character.width, character.height);
 }
 
+function drawPortal(){
```
Comment (deborahleehamel) [inline lib/index.js:22]: Possible portal sprites to review at: http://spritefx.blogspot.com/2013/04/sprite-portals.html

**Comment by bermannoah** [comment]
Should we go ahead and merge so we have it ready for next week?

**Comment by AliSchlereth** [comment]
Go for it!

**Pull request merged**